### PR TITLE
Add module support

### PR DESCRIPTION
**Pull request by katcipis**
To proper support modules I removed the readlines from the vendor since they are a modded/forked version of it, not just a vendor. Also needed to fix project root detection. Given how modules are implemented I think it is reasonably safe to use the go.mod file as an anchor to finding the root (you can't support modules without it).

Just cloning the project on any place in your file system, without a GOPATH defined, and running **make install** should work now.

### CODE REVIEW

**Comment by katcipis** [comment]
And it seems... that by magic travis integration is working again o.O

File: tests/cfg.go [hunk 1]
```diff
@@ -40,6 +33,58 @@ func init() {
...
+	// We used to use GOPATH as a way to infer the root of the
+	// project, now with Go modules this doesn't work anymore.
+	// Since module definition files only appear on the root
+	// of the project we use them instead, recursively going
+	// backwards in the file system until we find them.
+	//
+	// From: https://blog.golang.org/using-go-modules
+	// A module is a collection of Go packages stored in a file tree with a go.mod file at its root
+	//
+	// RIP GOPATH :-(
```
Comment (i4ki) [inline tests/cfg.go:52]: 😞

File: vendor/golang.org/x/exp/PATENTS [hunk 2]
```diff
@@ -0,0 +1,22 @@
+Additional IP Rights Grant (Patents)
```
Comment (i4ki) [inline vendor/golang.org/x/exp/PATENTS:1]: 🤔

File: vendor/github.com/chzyer/test/test.go [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+		return int64(n), true
+	case int32:
+		return int64(n), true
+	case int64:
+		return int64(n), true
+	case uintptr:
+		return int64(n), true
+	default:
+		return -1, false
+	}
```
Comment (i4ki) [inline vendor/github.com/chzyer/test/test.go:210]:
```
func toInt(a interface{}) (int64, bool) {
    switch n := a.(type) {
    case int, int8, int16, int32, int64, uintptr:
        return int64(n), true
    default:
        return -1, false
    }
}
```
🤣

File: tests/cfg.go [hunk 4]
```diff
@@ -40,6 +33,58 @@ func init() {
...
+
+	dir, err := os.Getwd()
+	separator := string(filepath.Separator)
+	if err != nil {
```
Comment (vitorarins) [inline tests/cfg.go:56]: Sorry if this is a stupid question, but why not check for the error immediately after `Getwd()`?

Comment (katcipis) [reply tests/cfg.go:56]: Because I'm stupid :-) (and your question is not, thanks :heart:

**Comment by katcipis** [comment]
> Looks like there is a lot of Windows stuff in readline we can get rid of in the future, since we're not supporting Windows anymore.

I just have a vague memory about this, did we really drop support for Windows @i4k ?

File: vendor/golang.org/x/exp/PATENTS (hunk 2 above)
Comment (katcipis) [reply vendor/golang.org/x/exp/PATENTS:1]: :thinking: :thinking:

File: tests/cfg.go (hunk 1 above)
Comment (katcipis) [reply tests/cfg.go:52]: It took me almost an year to accept that :-(. At least vendoring still works...kinda =P

**Comment by i4ki** [comment]
> > Looks like there is a lot of Windows stuff in readline we can get rid of in the future, since we're not supporting Windows anymore.
> 
> I just have a vague memory about this, did we really drop support for Windows @i4k ?

this is news to me but I don't care anymore.

**Comment by katcipis** [comment]
> > > Looks like there is a lot of Windows stuff in readline we can get rid of in the future, since we're not supporting Windows anymore.
> > 
> > 
> > I just have a vague memory about this, did we really drop support for Windows @i4k ?
> 
> this is news to me but I don't care anymore.

How careless =P

**Comment by i4ki** [comment]
> > > > Looks like there is a lot of Windows stuff in readline we can get rid of in the future, since we're not supporting Windows anymore.
> > > 
> > > 
> > > I just have a vague memory about this, did we really drop support for Windows @i4k ?
> > 
> > 
> > this is news to me but I don't care anymore.
> 
> How careless =P

you tagged the wrong i4k. ;-)

**Comment by katcipis** [comment]
> > > > > Looks like there is a lot of Windows stuff in readline we can get rid of in the future, since we're not supporting Windows anymore.
> > > > 
> > > > 
> > > > I just have a vague memory about this, did we really drop support for Windows @i4k ?
> > > 
> > > 
> > > this is news to me but I don't care anymore.
> > 
> > 
> > How careless =P
> 
> you tagged the wrong i4k. ;-)

:rofl:

**Comment by vitorarins** [comment]
> > > > > > Looks like there is a lot of Windows stuff in readline we can get rid of in the future, since we're not supporting Windows anymore.
> > > > > 
> > > > > 
> > > > > I just have a vague memory about this, did we really drop support for Windows @i4k ?
> > > > 
> > > > 
> > > > this is news to me but I don't care anymore.
> > > 
> > > 
> > > How careless =P
> > 
> > 
> > you tagged the wrong i4k. ;-)

:rofl:

**Comment by vitorarins** [comment]
> > > Looks like there is a lot of Windows stuff in readline we can get rid of in the future, since we're not supporting Windows anymore.
> > 
> > 
> > I just have a vague memory about this, did we really drop support for Windows @i4k ?
> 
> this is news to me but I don't care anymore.

Well, I also remember vaguely about the discussion we had offline about this.
But looking at the docs there is a section about environments where nash is tested, and those are all Linux.
Also we dropped OSX support in this PR #265 So for a moment I thought idea was to just focus on Linux

But anyway, was just an idea. Maybe we can bring back a proper support for Windows and OSX.

**Comment by i4ki** [comment]
We can drop windows for sure, but we need to bring back OSX.

**Comment by katcipis** [comment]
I think the release script still builds the binary for all platforms: 

https://github.com/madlambda/nash/blob/master/hack/releaser.sh#L9

Also the latest release has darwin/win:

https://github.com/madlambda/nash/releases/tag/v1.1

**Pull request merged**